js/functions.js

```diff
@@ -0,0 +1,91 @@
+
+const isStringShort = (testString, maxLength) => String(testString.length) <= maxLength;
+
+// Cтрока короче 20 символов
+console.log(isStringShort('проверяемая строка', 20)); // true
+// Длина строки ровно 18 символов
+console.log(isStringShort('проверяемая строка', 18)); // true
+// Строка длиннее 10 символов
+console.log(isStringShort('проверяемая строка', 10)); // false
+
+console.log('Задача 2');
+/* Функция для проверки, является ли строка палиндромом. Палиндром — это слово или фраза,
+которые одинаково читаются и слева направо и справа налево. */
+
+const isPalynodrome = (testString) => {
+  const string = String(testString.toLowerCase())
+    .replaceAll(' ', '');
+  const reverseString = string
+    .split('')
+    .reverse()
+    .join('');
+  return string === reverseString;
+};
+
+// Строка является палиндромом
+console.log(isPalynodrome('топот')); // true
+// Несмотря на разный регистр, тоже палиндром
+console.log(isPalynodrome('ДовОд')); // true
+// Это не палиндром
+console.log(isPalynodrome('Кекс'));  // false
+// Это палиндром
+console.log(isPalynodrome('Лёша на полке клопа нашёл ')); // true
+
+console.log('Задача 3');
+/* Функция, которая принимает строку, извлекает содержащиеся в ней цифры от 0 до 9
+и возвращает их в виде целого положительного числа.
+Если в строке нет ни одной цифры, функция должна вернуть NaN */
+
+/* const getNumbers = (testString) => {
+  let result = '';
+  for (let i = 0; i < testString.length; i++) {
+    if (!Number.isNaN(parseInt(testString.at(i), 10))) {
+      result += testString.at(i);
+    }
+  }
+  return parseInt(result, 10);
+
+  const string = testString.replaceAll(' ', '');
+  return string;
+}; */
+
+const getNumbers = (testString) => {
+  const string = String(testString)
+    .replace(/\D/g, '');
+  return (string === '') ? NaN : parseInt(string, 10);
+};
+
+console.log(getNumbers('2023 год'));            // 2023
+console.log(getNumbers('ECMAScript 2022'));     // 2022
+console.log(getNumbers('1 кефир, 0.5 батона')); // 105
+console.log(getNumbers('агент 007'));           // 7
+console.log(getNumbers('а я томат'));           // NaN
+console.log(getNumbers(2023));                  // 2023
+console.log(getNumbers(-1));                    // 1
+console.log(getNumbers(1.5));                   // 15
+
+console.log('Задача 4');
+/* Функция, которая принимает три параметра: исходную строку, минимальную длину и строку с добавочными символами —
+и возвращает исходную строку, дополненную указанными символами до заданной длины.
+Символы добавляются в начало строки. Если исходная строка превышает заданную длину, она не должна обрезаться.
+Если «добивка» слишком длинная, она обрезается с конца. */
+
+const getString = (testString, minLength, addlString) => {
+  const actualString = minLength - testString.length;
+  return (actualString <= 0) ? testString : addlString.slice(0, actualString % addlString.length) + addlString.repeat(actualString / addlString.length) + testString;
+};
```

Comment:
есть проблемы с нэймингом, во первых addlString - сокращений лучше избегай, потому что есть критерий [Б4.](https://up.htmlacademy.ru/javascript/28/criteries#b4)

во вторых actualString - название как будто в переменной строка, но по сути там число, поэтому лучше назвать как-то типа extraLength

ну и по поводу форматирования, можно переносить 
```suggestion
const getString = (testString, minLength, addlString) => {
  const actualString = minLength - testString.length;
  return (actualString <= 0) ? testString 
    : addlString.slice(0, actualString % addlString.length) 
    + addlString.repeat(actualString / addlString.length) + testString;
};
```

а вообще не обязательно везде писать именно тернарные операторы, в [Д17.](https://up.htmlacademy.ru/javascript/28/criteries#d17) речь именно про присваивания идёт , тут можно было и через if делать

Reply:
Done.